integration-test/integration/chat/hooks_test.clj

```diff
@@ -0,0 +1,382 @@
+                      :arguments (m/pred #(and (string? %)
+                                               (re-find #"\"recursive\":true" %)))}])}
+           (llm.mocks/get-req-body :tool-calling-0))))))
+
+(deftest lifecycle-hooks-order-test
+  (testing "sessionStart, chatStart, chatEnd, sessionEnd ordering"
+    (eca/start-process!)
+
+    (let [log-path (str h/default-root-project-path "/.eca/hooks-log.txt")]
+      (io/make-parents log-path)
+      (spit log-path "")
+
+      (eca/request!
+       (fixture/initialize-request
+        {:initializationOptions
+         (hooks-init-options
+          {"session-start" {:type "sessionStart"
+                            :actions [{:type "shell"
+                                       :shell "echo sessionStart >> .eca/hooks-log.txt"}]}
+           "chat-start" {:type "chatStart"
+                         :actions [{:type "shell"
+                                    :shell "printf 'chatStart:%s\\n' \"$(jq -r '.resumed')\" >> .eca/hooks-log.txt"}]}
+           "chat-end" {:type "chatEnd"
+                       :actions [{:type "shell"
+                                  :shell "echo chatEnd >> .eca/hooks-log.txt"}]}
+           "session-end" {:type "sessionEnd"
+                          :actions [{:type "shell"
+                                     :shell "echo sessionEnd >> .eca/hooks-log.txt"}]}})}))
+
+      (eca/notify! (fixture/initialized-notification))
+
+      (llm.mocks/set-case! :simple-text-0)
+
+      ;; Start new chat
+      (let [resp (eca/request! (fixture/chat-prompt-request
+                                {:model "openai/gpt-4.1"
+                                 :message "Hello"}))
+            chat-id (:chatId resp)]
+        (is (string? chat-id))
+
+        ;; Resume existing chat
+        (eca/request! (fixture/chat-prompt-request
+                       {:chat-id chat-id
+                        :model "openai/gpt-4.1"
+                        :message "Resume"}))
+
+        ;; Delete chat to trigger chatEnd.
+        (eca/request! [:chat/delete {:chat-id chat-id}]))
+
+      ;; Shutdown session to trigger sessionEnd.
+      (eca/request! (fixture/shutdown-request))
+      (eca/notify! (fixture/exit-notification))
+
+      (let [lines (->> (slurp log-path)
+                       string/split-lines
+                       (remove string/blank?))]
+        ;; Expected order:
+        ;; - sessionStart
+        ;; - chatStart:false (new chat, not resumed)
+        ;; - chatEnd
+        ;; - sessionEnd
+        (is (= ["sessionStart"
+                "chatStart:false"
+                "chatEnd"
+                "sessionEnd"]
+               lines))))))
+
+(deftest posttoolcall-receives-tool-response-test
+  (testing "postToolCall hook receives tool_response and tool_input after tool execution"
+    (eca/start-process!)
+
+    (llm.mocks/set-case! :tool-calling-0)
+
+    (let [log-path (str h/default-root-project-path "/.eca/posttool-log.txt")]
```

Comment:
Almost sure this is causing windows integration tests to fail, we should not concat paths with str, but use io/file or fs/file